datalad/distribution/create_sibling_github.py

```diff
@@ -165,17 +169,29 @@ def __call__(
             access_protocol='https',
             publish_depends=None,
             private=False,
-            dryrun=False):
+            dryrun=False,
+            dry_run=False):
         # this is an absolute leaf package, import locally to avoid
         # unnecessary dependencies
-        from datalad.support.github_ import _make_github_repos
+        from datalad.support.github_ import _make_github_repos_
+
+        if reponame != normalize_reponame(reponame):
+            raise ValueError('Invalid name for a GitHub project: {}'.format(
+                reponame))
 
         # what to operate on
         ds = require_dataset(
             dataset, check_installed=True, purpose='create GitHub sibling')
+
+        res_kwargs = dict(
+            action='create_sibling_github [dry-run]' if dryrun else
```

Comment:
> This prevents triggering result hooks, when actually nothing
> happened.

Ah, that makes a lot of sense.  I should make a note to think about this for other sites, including the in-flight `run` PR.

Reply:
Just a thought: since those records are intended for analysis/act-upon by outside code, I wonder if "dry-run" should not appear in `action` but rather appear in a dedicated key? Without that, it would require an even more arbitrary `.endswith(['dry-run'])` for outside code to tell which action was "real" and which was actually executed.
The counter-argument I guess could be: a " [dry-run]" flavor of action is like an action on its own.

Reply:
This is an alternative, but I think more complicated than needed. I indeed saw it as "it is not the same (or even any) action, when --dry-run prevents any actual intervention taking place". Adding another key to indicate "surprise, this isn't really what you were looking for", would imply that any user of result hooks now needs to anticipate this possibility and specifically test for this scenario (every time).

In contrast, I cannot really see an example of a result hook that is meant to run immediately after an action that did not actually happen (dry-run), because, by definition, it will not find the normal outcome.